lib/charms/mysql/v0/mysql.py

```diff
@@ -1674,35 +1685,63 @@ def is_cluster_in_cluster_set(self, cluster_name: str) -> Optional[bool]:
 
         return cluster_name in cs_status["clusters"]
 
-    def cluster_metadata_exists(self, from_instance: str) -> bool:
+    def cluster_metadata_exists(self, from_instance: Optional[str] = None) -> bool:
         """Check if this cluster metadata exists on database."""
-        check_cluster_metadata_commands = (
-            "result = session.run_sql(\"SHOW DATABASES LIKE 'mysql_innodb_cluster_metadata'\")",
-            "content = result.fetch_all()",
-            "if content:",
-            (
-                '  result = session.run_sql("SELECT cluster_name FROM mysql_innodb_cluster_metadata'
-                f".clusters where cluster_name = '{self.cluster_name}';\")"
-            ),
-            "  print(bool(result.fetch_one()))",
-            "else:",
-            "  print(False)",
+        if from_instance:
+            check_cluster_metadata_commands = (
+                "result = session.run_sql(\"SHOW DATABASES LIKE 'mysql_innodb_cluster_metadata'\")",
+                "content = result.fetch_all()",
+                "if content:",
+                (
+                    '  result = session.run_sql("SELECT cluster_name FROM mysql_innodb_cluster_metadata'
```

Comment:
possibly use the same `select from where where exists` as below

Reply:
addressed in https://github.com/canonical/mysql-operator/pull/615/commits/bded9d5176239dbe6d93df32eba88049aa853637